lib/utils.js

```diff
@@ -7,98 +10,82 @@ var assert = require("assert");
-      assert.ok(!result.database.driver_options[splitOption[0]]);
-
-      result.database.driver_options[splitOption[0]] = splitOption[1];
+    if (authString) {
+      return {
+        source: getName(),
+        user: authString.split(':')[0],
+        pass: authString.split(':')[1]
+      };
+    } else {
+      return {};
     }
   }
 
-  var host_port;
-  if (hostList.length > 1) {
-    result.database.host = [];
-    result.database.port = [];
-    for (i = 0; i < hostList.length; i++) {
-      host_port = hostList[i].split(":");
-      assert.ok(host_port[0] && host_port[0].length > 0);
-      result.database.host.push(host_port[0]);
-      if (host_port.length > 1) {
-        result.database.port.push(host_port[1]);
-      } else {
-        result.database.port.push('27017');
-      }
-    }
-  }
-  else {
-    host_port = hostList[0].split(":");
-    assert.ok(host_port[0] && host_port[0].length > 0);
-    result.database.host = host_port[0];
-    if (host_port.length > 1) {
-      result.database.port = host_port[1];
+  // Only the primary host string contains the "mongodb://" so we use this
+  // function to prepend it to others when passing to url.parse
+  function prependProtocol (str) {
+    if (str.indexOf('mongodb://') === -1) {
+      return 'mongodb://' + str;
     } else {
-      result.database.port = '27017';
+      return str;
     }
   }
 
+  var result = {
+    database: {
+      driver_options: getOptionsObject(),
+      auth: getAuthenticationObject(),
+      name: getName(),
+
+      // Use map since we can have multiple hosts and ports
+      host: parts.map((p) => url.parse(prependProtocol(p)).hostname),
+      port: parts.map((p) => url.parse(p).port || 27017)
+    }
+  };
 
-  //Verify the config to be returned
-  assert.ok(result.database.name.length > 0);
-  assert.ok(result.database.auth.user.length > 0);
-  assert.ok(result.database.auth.pass.length > 0);
+  // Verify a database name is given in the connection string
+  assert.ok(result.database.name, 'database name must be defined, e.g mongodb://localhost:27017/DATABASE_NAME');
 
+  // At least one host and port must exist
+  assert.notEqual(result.database.host.length, 0, 'at least one host must be specified in the mongodb connection string');
+  assert.notEqual(result.database.port.length, 0, 'at least one port must be specified in the mongodb connection string');
 
-  return result;
+  result.database.host.forEach((h) => assert(h, `host value of "${h}" is not valid. must be a non empty string`));
+  result.database.port.forEach((p) => assert(p, `port value of "${p}" is not valid. must be a non empty string`));
+
+  if (!env('FH_USE_LOCAL_DB').asBool()) {
+    // We can't assume authentication is being used if run locally
+    assert.ok(result.database.auth.user, 'mongodb username was missing in connection string');
+    assert.ok(result.database.auth.pass, 'mongodb password was missing in connection string');
+  }
+
+  return result
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Add a semicolon at the end of this statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ASemicolon)